recite/admin.py

```diff
@@ -55,89 +70,105 @@ class ReciterAdmin(admin.ModelAdmin):
+                    ayah_str = filename[3:6]
+                    try:
+                        surah_number = int(surah_str)
+                        ayah_number = int(ayah_str)
+                    except ValueError:
+                        print(f"Not normal format mp3 file {filename}")
+                    file_paths[surah_number][ayah_number] = filepath
 
         return file_paths
 
-    def get_ayah_object_from_row(self, row):
-        """Getting Ayah object for Recitation from csv file row"""
-        ayah_id = row.get('ayat')
-        surah_id = row.get('sura')
-
-        ayah = Ayah.objects.get(
-            number=ayah_id,
-            surah=surah_id,
-        )
-        return ayah
-
-    def get_file_audio_path(self, file_paths, ayah):
-        """Search in uploaded files for corresponding ayah"""
+    def get_segments_dict(self, encoding, segments_file):
+        """
+        Process csv file uploaded and return dict
+        segments_dict[surah][ayah] = [segments]
+        """
 
-        file_ayah_name_expected = str(ayah) + '.mp3'
-        file_ayah = None
-        for file_path in file_paths:
-            if file_ayah_name_expected == os.path.basename(file_path):
-                file_ayah = file_path
-                break
+        segments_dict = defaultdict(lambda: defaultdict(list))
 
-        if not file_ayah:
-            raise ValueError(
-                f"Not found Recitation audio file for ayah {ayah}")
-
-        return file_ayah
-
-    def read_segments_csv(self, segments_file, encoding):
-        """Return csv file as ordered dict"""
-        segments_file = io.TextIOWrapper(segments_file, encoding=encoding)
+        segments_file = io.TextIOWrapper(
+            segments_file, encoding=encoding)
         segments_file.seek(0)
-        reader = csv.DictReader(segments_file)
-        reader = sorted(reader, key=lambda x: int(x['sura']))
-
-        return reader
-
-    def get_unzipped_files_dir(self, audio_zip_file):
-        """Return temporary directory of unzipped files"""
-        audio_files = ZipFile(audio_zip_file, 'r')
+        csv_content_dict = csv.DictReader(segments_file)
+        csv_content_sorted = sorted(
+            csv_content_dict, key=lambda x: int(x['sura']))
 
-        temp_dir = tempfile.mkdtemp()
-        audio_files.extractall(temp_dir)
-
-        return temp_dir
+        for row in csv_content_sorted:
+            segments_list = json.loads(row.get("segments"))
+            segments_string = ",".join(
+                (f"{segment[2]}:{segment[3]}" for segment in segments_list)
+            )
+            surah_str = row.get('sura')
+            ayah_str = row.get('ayat')
+            try:
+                surah_number = int(surah_str)
+                ayah_number = int(ayah_str)
+            except ValueError:
+                print(f"Not normal format for csv at row {row}")
+            segments_dict[surah_number][ayah_number] = segments_string
```

Comment:
What will happen if you make the following change?
```suggestion
            segments_dict[surah_number][ayah_number] = segments_list
```

Reply:
ValidationError at /admin/recite/reciter/add/

['List must contain tuples only']